rct229/rules/section22/section22rule22.py

```diff
@@ -0,0 +1,81 @@
+from rct229.data_fns.table_3_5_3_fns import table_3_5_3_lookup
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.baseline_systems.baseline_system_util import HVAC_SYS
+from rct229.ruleset_functions.get_baseline_system_types import get_baseline_system_types
+from rct229.schema.config import ureg
+from rct229.utils.std_comparisons import std_equal
+
+APPLICABLE_SYS_TYPES = [
+    HVAC_SYS.SYS_7,
+    HVAC_SYS.SYS_8,
+    HVAC_SYS.SYS_11_1,
+    HVAC_SYS.SYS_11_2,
+    HVAC_SYS.SYS_12,
+    HVAC_SYS.SYS_13,
+    HVAC_SYS.SYS_7B,
+    HVAC_SYS.SYS_8B,
+    HVAC_SYS.SYS_11B,
+    HVAC_SYS.SYS_12B,
+]
+
+
+class Section22Rule22(RuleDefinitionListIndexedBase):
+    """Rule 22 of ASHRAE 90.1-2019 Appendix G Section 22 (Chilled water loop)"""
+
+    def __init__(self):
+        super(Section22Rule22, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section22Rule22.ChillerRule(),
+            index_rmr="baseline",
+            id="22-22",
+            description="The baseline chiller efficiencies shall be modeled at the minimum efficiency levels for full load, in accordance with Tables G3.5.3.",
+            rmr_context="ruleset_model_instances/0",
+            list_path="chillers[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        # create a list contains all HVAC systems that are modeled in the rmi_b
+        available_type_lists = [
+            hvac_type
+            for hvac_type in baseline_system_types_dict.keys()
+            if len(baseline_system_types_dict[hvac_type]) > 0
+        ]
+        return any(
+            [
+                available_type in APPLICABLE_SYS_TYPES
+                for available_type in available_type_lists
+            ]
+        )
+
+    class ChillerRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section22Rule22.ChillerRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={
+                    "$": ["compressor_type", "rated_capacity"],
+                },
+            )
+
+        def get_calc_vals(self, context, data=None):
+            chiller_b = context.baseline
+            compressor_type_b = chiller_b["compressor_type"]
+            rated_capacity_b = chiller_b["rated_capacity"].to(ureg.ton).m
+
+            kW_ton_full_load_b = table_3_5_3_lookup(
+                compressor_type_b, rated_capacity_b
+            )["minimum_full_load_efficiency_kw_per_ton"]
+
+            return {
+                "kW_ton_full_load_b": kW_ton_full_load_b,
```

Comment:
Should use CalcQ class in here.
The `kw_ton_full_load_b` is table value from standard and it is the minimum requirement. So we should relabel this to `required_kw_ton_full_load_b`.